include/mbgl/style/filter.hpp

```diff
@@ -232,6 +233,15 @@ class NotHasIdentifierFilter {
         return true;
     }
 };
+    
+class ExpressionFilter {
+public:
+    std::shared_ptr<expression::Expression> expression;
+    
+    friend bool operator==(const ExpressionFilter& lhs, const ExpressionFilter& rhs) {
+        return lhs.expression == rhs.expression;
```

Comment:
Are you sure? I can't find a definitive answer but Googling around seems to indicate C++ will use `ExpressionFilter::operator==` rather than an address check.

Reply:
Pretty sure -- according to http://en.cppreference.com/w/cpp/memory/shared_ptr/operator_cmp, `shared_ptr::operator==` simply compares the wrapped `Expression*` pointers, as opposed to invoking `Expression::operator==()`.  Confirmed here: http://cpp.sh/7yskf

Reply:
Oh, doh, I was thinking about the `ExpressionFilter&`s, not the `shared_ptr<Expression>`s 😄